### PR TITLE
Mapstruct integration

### PR DESCRIPTION
**Pull request by holkwen**

### CODE REVIEW

File: src/main/java/net/greatstart/dto/DtoProject.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package net.greatstart.dto;
+
+import lombok.Data;
+
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+
+@Data
+public class DtoProject {
+    private Long id;
```
Comment (Sheptytskyid) [inline src/main/java/net/greatstart/dto/DtoProject.java]: MAJOR Remove this unused "id" private field. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)

File: src/main/java/net/greatstart/dto/DtoCategory.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package net.greatstart.dto;
+
+import lombok.Data;
+
+@Data
+public class DtoCategory {
+    private String name;
```
Comment (Sheptytskyid) [inline src/main/java/net/greatstart/dto/DtoCategory.java]: MAJOR Remove this unused "name" private field. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)

File: src/main/java/net/greatstart/dto/DtoProjectDescription.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import javax.validation.constraints.Size;
+import java.math.BigDecimal;
+import java.time.LocalDate;
+
+@Data
+public class DtoProjectDescription {
+    @NotNull
+    @Size(min = 3, max = 35, message = "{name.size}")
+    private String name;
+    private String goal;
```
Comment (Sheptytskyid) [inline src/main/java/net/greatstart/dto/DtoProjectDescription.java]: MAJOR Remove this unused "goal" private field. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)

File: src/main/java/net/greatstart/dto/DtoProjectDescription.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    private String description;
```
Comment (Sheptytskyid) [inline src/main/java/net/greatstart/dto/DtoProjectDescription.java]: MAJOR Remove this unused "description" private field. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)

File: src/main/java/net/greatstart/dto/DtoProjectDescription.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    @DecimalMin(value = "1000.00", message = "{project.cost.min}")
+    private BigDecimal cost;
+    private BigDecimal minInvestment;
```
Comment (Sheptytskyid) [inline src/main/java/net/greatstart/dto/DtoProjectDescription.java]: MAJOR Remove this unused "minInvestment" private field. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)

File: src/main/java/net/greatstart/dto/DtoProjectDescription.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    private LocalDate dateStart;
```
Comment (Sheptytskyid) [inline src/main/java/net/greatstart/dto/DtoProjectDescription.java]: MAJOR Remove this unused "dateStart" private field. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)

File: src/main/java/net/greatstart/dto/DtoProjectDescription.java [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+    private LocalDate dateAdded;
```
Comment (Sheptytskyid) [inline src/main/java/net/greatstart/dto/DtoProjectDescription.java]: MAJOR Remove this unused "dateAdded" private field. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)

File: src/main/java/net/greatstart/dto/DtoProjectDescription.java [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+    private byte[] image;
```
Comment (Sheptytskyid) [inline src/main/java/net/greatstart/dto/DtoProjectDescription.java]: MAJOR Remove this unused "image" private field. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)

**Comment by Sheptytskyid** [comment]
SonarQube analysis reported 8 issues
* ![MAJOR][MAJOR] 8 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: src/main/java/net/greatstart/controllers/ProjectController.java [hunk 9]
```diff
@@ -25,15 +33,25 @@
...
     private static final String REDIRECT_TO_PROJECTS = "redirect:/project/";
     private static final String PROJECTS = "project/projects";
-    private static final String PROJECT = "project";
+    private static final String PROJECT = "dtoProject";
+
+    private final Logger logger = LoggerFactory.getLogger(ProjectController.class);
 
     private ProjectService projectService;
     private UserService userService;
+    private CategoryService categoryService;
```
Comment (Sheptytskyid) [inline src/main/java/net/greatstart/controllers/ProjectController.java:42]: MAJOR Remove this unused "categoryService" private field. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)

**Comment by Sheptytskyid** [comment]
SonarQube analysis reported 9 issues
* ![MAJOR][MAJOR] 9 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: src/main/java/net/greatstart/controllers/UserController.java [hunk 10]
```diff
@@ -31,13 +31,13 @@
...
 
     @Autowired
-    public UserController(UserService userService, UserConverterService userConverterService) {
+    public UserController(UserService userService, UserProfileMapper userMapper) {
         this.userService = userService;
-        this.userConverter = userConverterService;
+        this.userMapper = userMapper;
     }
 
     // TODO error pages (no such user etc.); enable admin user to edit profiles
```
Comment (Sheptytskyid) [inline src/main/java/net/greatstart/controllers/UserController.java]: INFO Complete the task associated to this TODO comment. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1135)

**Comment by Sheptytskyid** [comment]
SonarQube analysis reported 2 issues
* ![MAJOR][MAJOR] 1 major
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: src/main/java/net/greatstart/dto/DtoProject.java (hunk 1 above)
Comment (Sheptytskyid) [inline src/main/java/net/greatstart/dto/DtoProject.java:10]: MAJOR Remove this unused "id" private field. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)

File: src/main/java/net/greatstart/dto/DtoCategory.java (hunk 2 above)
Comment (Sheptytskyid) [inline src/main/java/net/greatstart/dto/DtoCategory.java:7]: MAJOR Remove this unused "name" private field. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)

File: src/main/java/net/greatstart/dto/DtoProjectDescription.java (hunk 3 above)
Comment (Sheptytskyid) [inline src/main/java/net/greatstart/dto/DtoProjectDescription.java:16]: MAJOR Remove this unused "goal" private field. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)

File: src/main/java/net/greatstart/dto/DtoProjectDescription.java (hunk 4 above)
Comment (Sheptytskyid) [inline src/main/java/net/greatstart/dto/DtoProjectDescription.java:17]: MAJOR Remove this unused "description" private field. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)

File: src/main/java/net/greatstart/dto/DtoProjectDescription.java (hunk 5 above)
Comment (Sheptytskyid) [inline src/main/java/net/greatstart/dto/DtoProjectDescription.java:20]: MAJOR Remove this unused "minInvestment" private field. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)

File: src/main/java/net/greatstart/dto/DtoProjectDescription.java (hunk 6 above)
Comment (Sheptytskyid) [inline src/main/java/net/greatstart/dto/DtoProjectDescription.java:21]: MAJOR Remove this unused "dateStart" private field. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)

File: src/main/java/net/greatstart/dto/DtoProjectDescription.java (hunk 7 above)
Comment (Sheptytskyid) [inline src/main/java/net/greatstart/dto/DtoProjectDescription.java:22]: MAJOR Remove this unused "dateAdded" private field. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)

File: src/main/java/net/greatstart/dto/DtoProjectDescription.java (hunk 8 above)
Comment (Sheptytskyid) [inline src/main/java/net/greatstart/dto/DtoProjectDescription.java:23]: MAJOR Remove this unused "image" private field. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)

**Comment by Sheptytskyid** [comment]
SonarQube analysis reported 9 issues
* ![MAJOR][MAJOR] 9 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: src/test/java/net/greatstart/controllers/EventControllerTest.java [hunk 11]
```diff
@@ -0,0 +1,23 @@
+package net.greatstart.controllers;
+
+import net.greatstart.services.EventService;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
```
Comment (Sheptytskyid) [inline src/test/java/net/greatstart/controllers/EventControllerTest.java]: BLOCKER Add some tests to this class. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS2187)

**Comment by Sheptytskyid** [comment]
SonarQube analysis reported 10 issues
* ![BLOCKER][BLOCKER] 1 blocker
* ![MAJOR][MAJOR] 9 major

Watch the comments in this conversation to review them.

[BLOCKER]: https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by Sheptytskyid** [comment]
SonarQube analysis reported 9 issues
* ![MAJOR][MAJOR] 9 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Review by zozich — APPROVED** [review]
👍

**Pull request merged**